src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -89,78 +89,43 @@ def on_simulation_end(self, event: Event) -> None:
-    save population table with / key = date
-    """
-
-    def __repr__(self) -> str:
-        return "Observers()"
-
-    ##############
-    # Properties #
-    ##############
+    def __repr__(self):
+        return f"DecennialCensusObserver()"
 
     @property
     def name(self):
-        return "observers"
-
-    #################
-    # Setup methods #
-    #################
+        return f"decennial_census_observer"
+    
+    @property
+    def output_filename(self):
+        return f"decennial_census.hdf"
 
     def setup(self, builder: Builder):
-        self.start_date = builder.configuration.time.start
-        self.end_date = builder.configuration.time.end
+        super().setup(builder)
         self.clock = builder.time.clock()
-        self.counter = 0
-        # FIXME: Are these the correct output locations?
-        self.output_path = build_output_dir(builder, subdir="population_table") / "state_table.hdf"
-        self.decennial_path = build_output_dir(builder, subdir="population_table") / "decennial_census.hdf"
-
-        self.randomness = builder.randomness.get_stream(self.name)
-        self.population_view = builder.population.get_view(columns=[])
-        self.response_probability_decennial = builder.lookup.build_table(
-            data=data_values.RESPONSE_PROBABILITY_DECENNIAL
-        )
+        
+    def get_population_view(self, builder) -> PopulationView:
+        """Get the population view to be used for observations"""
+        return builder.population.get_view(columns=metadata.DECENNIAL_CENSUS_COLUMNS_USED)
 
-        # FIXME: register to happen "on_collect_metrics" (end of time step)
-        builder.event.register_listener("time_step__prepare", self.on_time_step__prepare)
-        builder.event.register_listener("simulation_end", self.on_simulation_end)
-
-    def on_time_step__prepare(self, event: Event) -> None:
-        if self.counter == 0:
-            start_date_str = (
-                f"ymd_{self.start_date.year}_{self.start_date.month}_{self.start_date.day}"
-            )
-            state_table = self.population_view.get(event.index)
-            state_table.to_hdf(self.output_path, start_date_str)
-        self.counter += 1
-        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
-            if self.clock().day < 29:  # because we only want one observation in April
-                self.decennial_census(event, hdf_key=f"year_{self.clock().year}")
+    def get_responses(self) -> pd.DataFrame:
+        return pd.DataFrame()  # TODO: include schema here, including column for census year
 
-    def on_simulation_end(self, event: Event) -> None:
-        end_date_str = f"ymd_{self.end_date.year}_{self.end_date.month}_{self.end_date.day}"
-        state_table = self.population_view.get(event.index)
-        state_table.to_hdf(self.output_path, end_date_str)
+    def to_observe(self, event: Event) -> bool:
+        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
```

Comment:
You should be using `event.time` not `self.clock()`. The previous code was incorrect.

Is there a reason you are using `&` rather than `and`?

This observer should return `False` when we're not supposed to observe. Currently it returns `None` which means it will work, since `None` is falsey, but this is bad practice. It's much better in a boolean function like this just to return the conditional you currently have in an `if` statement like I noted in my above comment.

Reply:
I am trying to make this PR simply a refactor of the existing observer, and I would prefer to improve this logic in a future PR --- The logic here is identical to what was present before. I can improve it here if you want, but I think it will be clearer to keep the shift to BaseObserver separate from the other changes.